pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -27,10 +27,277 @@ import (
+	// storage through different nodes within a cohort
+	Multipathing StorageCohortMultipathingOption `json:"multipathing,omitempty"`
+
+	// If specified, the compression algorithm supported by the cohort
+	// +nullable
+	Compression *[]StorageCohortCompressionAlgorithm `json:"compression,omitempty"`
+
+	// Deduplication, if set to true, ensures that the data is de-duplicated within the cohort
+	// +nullable
+	Deduplication *bool `json:"deduplication,omitempty"`
+}
+
+// StorageCohortDataSecurityCapabilities
+type StorageCohortDataSecurityCapabilities struct {
+	// MediaEncryption either specifies whether encryption is supported or specifies the encryption algorithm itself
+	// +nullable
+	MediaEncryption *[]StorageCohortEncryptionAlgorithm `json:"mediaEncryption,omitempty"`
+
+	// DataSanitizationPolicy specifies the supported data sanity checks
+	DataSanitizationPolicy StorageCohortDataSanitizationPolicy `json:"dataSanitizationPolicy,omitempty"`
+}
+
+// StorageCohortIOConnectivityCapabilities
+type StorageCohortIOConnectivityCapabilities struct {
+	// Contains the list of access protocols supported by the cohort
+	AccessProtocols []StorageCohortAccessProtocol `json:"accessProtocols,omitempty"`
+}
+
+// StorageCohortIOPerformanceCapabilities
+type StorageCohortIOPerformanceCapabilities struct {
+	// Average I/O operation latency(in microseconds) allowed within the cohort
+	AverageIOOperationLatencyMicroseconds uint64 `json:"averageIoOperationLatencyMicroseconds,omitempty"`
+
+	// Max I/O operation(per second/TB) allowed within the cohort
+	MaxIOOperationsPerSecondPerTerabyte uint64 `json:"maxIoOperationsPerSecondPerTerabyte,omitempty"`
+
+	// List of storage tiers supported by the cohort
+	StorageTier []StorageCohortStorageTier `json:"storageTier,omitempty"`
 }
 
+// StorageCohortDataProtectionCapabilities
+type StorageCohortDataProtectionCapabilities struct {
+	// Whether replication is supported by the cohort
+	// +optional
+	Replication *bool `json:"replication,omitempty"`
+
+	// Backup controls data can be backed up by the cohort
+	// +optional
+	Backup *bool `json:"backup,omitempty"`
+
+	// whether cohort will support snapshots
+	// +optional
+	Snapshots *bool `json:"snapshots,omitempty"`
+}
+
+// ComponentCondition contains condition information for a storage cohort.
+type ComponentCondition struct {
+	// Name of the component
+	// For example: "cohort-manager"
+	Name string `json:"name"`
+	// Type of component condition.
+	Type ComponentConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status ComponentConditionStatus `json:"status"`
+	// Last time we got an update on a given condition.
+	LastHeartbeatTime metav1.Time `json:"lastHeartbeatTime,omitempty"`
+	// Last time the condition transit from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// (brief) reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
```

Comment:
What is difference between Reason and Message

Reply:
Reason can be like: `CohortReady` and message can be: `Cohort is posting ready status. Authorization enabled`. Something like this.